wgpu-types/src/lib.rs

```diff
@@ -1303,6 +1358,49 @@ impl<L> TextureDescriptor<L> {
     }
 }
 
+/// Texture descriptor builder methods
+impl<'l> TextureDescriptor<Option<Cow<'l, str>>> {
+    pub fn new(
+        size: Extent3d,
+        dimension: TextureDimension,
+        format: TextureFormat,
+        usage: TextureUsage,
+    ) -> Self {
+        Self {
+            label: None,
+            size,
+            mip_level_count: 1,
+            sample_count: 1,
+            dimension,
+            format,
+            usage,
+        }
+    }
+
+    pub fn build(&self) -> Self {
+        Self::clone(self)
+    }
+
+    pub fn label<L>(&mut self, label: L) -> &mut Self
```

Comment:
given that we are going to use this pattern a lot (a generic that is into-cowed), could we use the shorter syntax for this? I.e. `pub fn label(&mut self, label: impl IntoCow<'l, CowType = str>) -> &mut Self`

Reply:
Yeah, agreed. I thought I needed a manual where, but it doesn't turn out needed.